File: Projekt/src/main/java/de/neosit/minecraft/arduinointegration/commands/ArduinoCommand.java
package de.neosit.minecraft.arduinointegration.commands;

import de.neosit.minecraft.arduinointegration.Main;
import org.apache.commons.lang.ArrayUtils;
import org.apache.commons.lang.StringUtils;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.Arrays;
import java.util.List;

public class ArduinoCommand implements CommandExecutor {
    private Main main;

    public ArduinoCommand(Main main) {
        this.main = main;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        List<String> helpCmds = Arrays.asList("-h", "--help", "?");
        if (args.length == 1 && helpCmds.contains(args[0])) {
            sender.sendMessage(command.getUsage());
            return true;
        }

        if (args.length > 0) {
            switch (args[0]) {
                case "send":
                    if (args.length == 1) {
                        sender.sendMessage(main.getCommand("arduino send").getDescription());
                        return true;
                    }
                    String message = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
                    main.getSerial().sendData(message);
                    return true;
                case "reload":
                    if (args.length == 1) {
                        sender.sendMessage(main.getCommand("arduino reload").getDescription());
                        return true;
                    }
                    if (args.length == 2 && args[1].equals("config")) {
                        main.reloadConfig();
                        main.getConfiguration().reload(main.getConfig());
                        return true;
                    }
            }
        }

        // all other cases
        sender.sendMessage(
                "Unknown command \"" + StringUtils.join(args, " ") + "\". " +
                        "Type " + ChatColor.BLUE + "/arduino --help" + ChatColor.RESET + " for all available commands.");
        return true;
    }
}

File: Projekt/src/main/java/de/neosit/minecraft/arduinointegration/commands/ArduinoCommand.kt
package de.neosit.minecraft.arduinointegration.commands

import de.neosit.minecraft.arduinointegration.Main
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class ArduinoCommand(private val main: Main) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        when {
            args.size == 1 && helpCmds.contains(args[0]) -> {
                sender.sendMessage(command.usage)
                return true
            }
            args.isEmpty() -> {
                sender.sendMessage("""
                    Unknown command "${args.joinToString(" ")}"
                    Type ${ChatColor.BLUE}/arduino --help${ChatColor.RESET} for all available commands.
                """.trimIndent())
                return true
            }
            args[0] == "send" && args.size == 1 -> {
                sender.sendMessage(main.getCommand("arduino send").description)
                return true
            }
            args[0] == "send" -> {
                val message = args.slice(1 until args.count()).joinToString(" ")
                main.serial.sendData(message)
                return true
            }
            args[0] == "reload" && args.size == 1 -> {
                sender.sendMessage(main.getCommand("arduino reload").description)
                return true
            }
            args[0] == "reload" && args.size == 2 && args[1] == "config" -> {
                main.reloadConfig()
                main.configuration.reload(main.config)
                return true
            }
            else -> {
                return false
            }
        }
    }
}

val helpCmds = listOf("-h", "--help", "?")
